src/coreclr/src/jit/liveness.cpp

```diff
@@ -342,16 +342,16 @@ void Compiler::fgPerNodeLocalVarLiveness(GenTree* tree)
                 fgCurMemoryDef |= memoryKindSet(GcHeap, ByrefExposed);
                 fgCurMemoryHavoc |= memoryKindSet(GcHeap, ByrefExposed);
             }
-        }
 
-            // If this is a p/invoke unmanaged call or if this is a tail-call
+            // If this is a p/invoke unmanaged call or if this is a tail-call via helper,
             // and we have an unmanaged p/invoke call in the method,
             // then we're going to run the p/invoke epilog.
             // So we mark the FrameRoot as used by this instruction.
             // This ensures that the block->bbVarUse will contain
             // the FrameRoot local var if is it a tracked variable.
 
-            if ((tree->AsCall()->IsUnmanaged() || tree->AsCall()->IsTailCall()) && compMethodRequiresPInvokeFrame())
+            if ((tree->AsCall()->IsUnmanaged() || tree->AsCall()->IsTailCallViaJitHelper()) &&
+                compMethodRequiresPInvokeFrame())
```

Comment:
Just below this is another comment mentioning "the TCB". Since you redid other comments, maybe redo this one too?